cms/djangoapps/contentstore/management/commands/tests/test_migrate_transcripts.py

```diff
@@ -105,9 +106,9 @@ def setUp(self):
                    youtube="1.0:p2Q6BrNhdh8,0.75:izygArpw-Qo,1.25:1EeWXzPdhSA,1.5:rABDYkeK0x8"
                    show_captions="false"
                    download_track="false"
-                   start_time="00:00:01"
+                   start_time="1.0"
                    download_video="false"
-                   end_time="00:01:00">
+                   end_time="60.0">
```

Comment:
Ah yes, this was a tricky and I tried a number of ways to get it to work.

The problem is that `deserialize_fields` gets [called](https://github.com/edx/edx-platform/blob/master/common/lib/xmodule/xmodule/xml_module.py#L83) on this value when [setting](https://github.com/edx/edx-platform/blob/master/common/lib/xmodule/xmodule/video_module/video_module.py#L958) it in the field data. However, `json.loads()` fails because it is not a valid json value so the value that gets put is a string and not a time object.

This causes this error when trying to save the XBlock:
```
lms/djangoapps/courseware/tests/test_video_handlers.py:142: in initialize_block
    super(BaseTestVideoXBlock, self).initialize_module(**kwargs)
lms/djangoapps/courseware/tests/helpers.py:81: in initialize_module
    self.item_descriptor = ItemFactory.create(**kwargs)
../venvs/edxapp/local/lib/python2.7/site-packages/factory/base.py:623: in create
    return cls._generate(True, attrs)
../venvs/edxapp/local/lib/python2.7/site-packages/factory/base.py:548: in _generate
    obj = cls._prepare(create, **attrs)
../venvs/edxapp/local/lib/python2.7/site-packages/factory/base.py:523: in _prepare
    return cls._create(model_class, *args, **kwargs)
common/lib/xmodule/xmodule/modulestore/tests/factories.py:389: in _create
    fields=kwargs,
lms/djangoapps/ccx/modulestore.py:212: in create_child
    user_id, parent_usage_key, block_type, block_id=block_id, fields=fields, **kwargs
common/lib/xmodule/xmodule/modulestore/mixed.py:93: in inner
    retval = func(field_decorator=strip_key_collection, *args, **kwargs)
common/lib/xmodule/xmodule/modulestore/mixed.py:113: in wrapper
    return func(*args, **kwargs)
common/lib/xmodule/xmodule/modulestore/mixed.py:775: in create_child
    return modulestore.create_child(user_id, parent_usage_key, block_type, block_id=block_id, fields=fields, **kwargs)
common/lib/xmodule/xmodule/modulestore/mongo/base.py:1452: in create_child
    xblock = self.create_item(user_id, parent_usage_key.course_key, block_type, block_id=block_id, **kwargs)
common/lib/xmodule/xmodule/modulestore/mongo/base.py:1426: in create_item
    xblock = self.create_xblock(runtime, course_key, block_type, block_id, **kwargs)
common/lib/xmodule/xmodule/modulestore/mongo/draft.py:337: in create_xblock
    runtime, course_key, block_type, block_id, fields, **kwargs
common/lib/xmodule/xmodule/modulestore/mongo/base.py:1402: in create_xblock
    xmodule.save()
../venvs/edxapp/local/lib/python2.7/site-packages/xblock/mixins.py:255: in save
    self.force_save_fields(fields_to_save)
../venvs/edxapp/local/lib/python2.7/site-packages/xblock/mixins.py:265: in force_save_fields
    fields_to_save_json[field.name] = field.to_json(self._field_data_cache[field.name])
```

As far as I understand this issue only impacts the tests because the `start` and `end` values when loaded from `data` are not considered changed and so not saved during `xmodule.save()`.

@ormsbee Do you have any thoughts on this?

Reply:
That is painful to read, but given the circumstances, I think what you're doing is fine.